zcash_client_backend/src/proposal.rs

```diff
@@ -98,21 +98,127 @@ impl<NoteRef> ShieldedInputs<NoteRef> {
     }
 }
 
-/// The inputs to be consumed and outputs to be produced in a proposed transaction.
+/// A proposal for a series of transactions to be created.
+///
+/// Each step of the proposal represents a separate transaction to be created. At present, only
+/// transparent outputs of earlier steps may be spent in later steps; the ability to chain shielded
+/// transaction steps may be added in a future update.
 #[derive(Clone, PartialEq, Eq)]
 pub struct Proposal<FeeRuleT, NoteRef> {
+    fee_rule: FeeRuleT,
+    min_target_height: BlockHeight,
+    steps: Vec<Step<NoteRef>>,
+}
+
+impl<FeeRuleT, NoteRef> Proposal<FeeRuleT, NoteRef> {
+    /// Constructs a validated multi-step [`Proposal`].
+    ///
+    /// This operation validates the proposal for agreement between outputs and inputs
+    /// in the case of multi-step proposals, and ensures that no double-spends are being
+    /// proposed.
+    ///
+    /// Parameters:
+    /// * `fee_rule`: The fee rule observed by the proposed transaction.
+    /// * `min_target_height`: The minimum block height at which the transaction may be created.
+    /// * `steps`: A vector of steps that make up the proposal.
+    pub fn multi_step(
+        fee_rule: FeeRuleT,
+        min_target_height: BlockHeight,
+        steps: Vec<Step<NoteRef>>,
+    ) -> Result<Self, ProposalError> {
+        // TODO: actually perform the validation described in the documentation.
```

Comment:
If being left as a TODO, please open a downstream issue for this that blocks the appropriate release.

Reply:
Fixed in 063c472